tests/std/tests/GH_002307_usual_scope_guard/test.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#define _CORE_HEADERS_ONLY
+
+#include <cassert>
+#include <deque>
+#include <stdexcept>
+
+using namespace std;
+
+struct bomb {
+    int val;
+
+    static int countdown;
+
+    void tick() {
+        if (countdown == 0) {
+            throw runtime_error("BOOM");
+        } else {
+            --countdown;
+        }
+    }
+
+    bomb(int init) : val(init) {
+        tick();
+    }
+
+    bomb(const bomb& other) : val(other.val) {
+        tick();
+    }
+
+    bomb& operator=(const bomb& other) {
+        val = other.val;
+        return *this;
+    }
+
+    bool operator==(int other) const {
+        return val == other;
+    }
+};
+
+int bomb::countdown = 0;
+
+
+int main() {

```

Comment:
This will most likely also be implemented for other containers, so lets go wit 

```suggestion
void test_deque() {
    deque<bomb> dq(begin(data), end(data));
    test<Operation::InsertRangeFront>(dq);
    test<Operation::InsertRangeBack>(dq);
    test<Operation::InsertCountFront>(dq);
    test<Operation::InsertCountBack>(dq);
}

int main() {
```

I am not really sure whether it would be cleaner to simply create the container inside the `test` function and not pass it around. It is slightly more efficient but I doubt that it is really measurable

Reply:
I'm not sure. Depends whether `_RERAISE` is used there the same way. Let's defer it until other containers.

Reply:
We use it also in vector and xtree, so essentially every container

Reply:
The usages are different, and it would take different test to test it

Reply:
We could also pass a lambda, to show what is done and factor out the common boilerplate
```cpp
test(dq, [&dq](){ dq.insert(dq.begin() + 2, begin(more_data), end(more_data)); })
```

Reply:
This test will not give proper coverage for xtree and vector. It won't even compile for tree-based containers. So there's a little point to reuse it.

Reply:
Thanks @miscco for thinking about future work in this area. I agree with @AlexGuteniev that such refactoring should be deferred until the other containers are updated in the same way - then we'll be able to identify what, if any, commonality can be extracted while still having clear test code.

Reply:
Resolving this as the test now has `test_deque()`, indicating that future additions are expected. Please meow if you disagree.